airbyte-integrations/connectors/source-mysql/src/main/kotlin/io/airbyte/integrations/source/mysql/MySqlSourceConfiguration.kt

```diff
@@ -101,57 +106,35 @@ class MySqlSourceConfigurationFactory @Inject constructor(val featureFlags: Set<
-            when (encryption) {
-                is EncryptionPreferred -> {
-                    if (
-                        featureFlags.contains(FeatureFlag.AIRBYTE_CLOUD_DEPLOYMENT) &&
-                            sshTunnel is SshNoTunnelMethod
-                    ) {
-                        throw ConfigErrorException(
-                            "Connection from Airbyte Cloud requires " +
-                                "SSL encryption or an SSH tunnel."
-                        )
-                    }
-                    MySqlSourceEncryption(sslMode = MySqlSourceEncryption.SslMode.PREFERRED)
-                }
-                is EncryptionRequired ->
-                    MySqlSourceEncryption(sslMode = MySqlSourceEncryption.SslMode.REQUIRED)
-                is SslVerifyCertificate ->
-                    MySqlSourceEncryption(
-                        sslMode = MySqlSourceEncryption.SslMode.VERIFY_CA,
-                        caCertificate = encryption.sslCertificate,
-                        clientCertificate = encryption.sslClientCertificate,
-                        clientKey = encryption.sslClientKey,
-                        clientKeyPassword = encryption.sslClientPassword
-                    )
-                is SslVerifyIdentity ->
-                    MySqlSourceEncryption(
-                        sslMode = MySqlSourceEncryption.SslMode.VERIFY_IDENTITY,
-                        caCertificate = encryption.sslCertificate,
-                        clientCertificate = encryption.sslClientCertificate,
-                        clientKey = encryption.sslClientKey,
-                        clientKeyPassword = encryption.sslClientPassword
-                    )
-                null -> TODO()
-            }
-        val sslJdbcParameters = jdbcEncryption.parseSSLConfig()
-        jdbcProperties.putAll(sslJdbcParameters)
-
-        val cursorConfig = pojo.getCursorMethodConfigurationValue()
-        val maxSnapshotReadTime: Duration? =
-            when (cursorConfig is CdcCursor) {
-                true -> cursorConfig.initialLoadTimeoutHours?.let { Duration.ofHours(it.toLong()) }
-                else -> null
-            }
-        // Build JDBC URL
+
+        // Configure SSH tunneling.
+        val sshTunnel: SshTunnelMethodConfiguration? = pojo.getTunnelMethodValue()
+        val sshOpts: SshConnectionOptions =
+            SshConnectionOptions.fromAdditionalProperties(pojo.getAdditionalProperties())
+
+        // Configure SSL encryption.
+        if (
+            pojo.getEncryptionValue() is EncryptionPreferred &&
+                sshTunnel is SshNoTunnelMethod &&
+                featureFlags.contains(FeatureFlag.AIRBYTE_CLOUD_DEPLOYMENT)
+        ) {
+            throw ConfigErrorException(
+                "Connection from Airbyte Cloud requires SSL encryption or an SSH tunnel."
+            )
+        }
+        val sslJdbcProperties: Map<String, String> = fromEncryptionSpec(pojo.getEncryptionValue()!!)
```

Comment:
nit: can we make getEncryptionValue() non-null?

Reply:
Unfortunately not, because for legacy reasons the json schema of the config declares this field as optional instead of specifying a default value. Basically I have to set this as nullable for the json schema generation to do its thing and not consider existing config jsons as invalid. Sad.

Reply:
why can't we make it non-nullable with a default of "encryptionPrefered" and ignore set(null) ? That way, the handling of legacy configs is handled with the config.